rest/models/snapshot.go

```diff
@@ -378,3 +380,97 @@ type OrderBookQuote struct {
 	Price            float64            `json:"p,omitempty"`
 	ExchangeToShares map[string]float64 `json:"x,omitempty"`
 }
+
+type ListAssetSnapshotsParams struct {
+	TickerAnyOf *string `query:"ticker.any_of"`
+	Ticker      *string `query:"ticker"`
+
+	TickerLT  *string `query:"ticker.lt"`
+	TickerLTE *string `query:"ticker.lte"`
+	TickerGT  *string `query:"ticker.gt"`
+	TickerGTE *string `query:"ticker.gte"`
+
+	Type *string `query:"type"`
+}
+
+func (p ListAssetSnapshotsParams) WithTickerAnyOf(q string) *ListAssetSnapshotsParams {
+	p.TickerAnyOf = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithTicker(q string) *ListAssetSnapshotsParams {
+	p.Ticker = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithType(q string) *ListAssetSnapshotsParams {
+	p.Type = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithTickersByComparison(c Comparator, q string) *ListAssetSnapshotsParams {
+	switch c {
+	case LT:
+		p.TickerLT = &q
+	case LTE:
+		p.TickerLTE = &q
+	case GT:
+		p.TickerGT = &q
+	case GTE:
+		p.TickerGTE = &q
+	}
+	return &p
+}
+
+type ListAssetSnapshotsResponse struct {
+	BaseResponse
+	Results []SnapshotResponseModel `json:"results,omitempty"`
+}
+
+type SnapshotResponseModel struct {
```

Comment:
These may have mistakenly been left out of the response model which the API defines, but each result can also be an error object and look like this: 
```
    {
      "error": "NOT_FOUND",
      "message": "Ticker not found.",
      "ticker": "TSLAAPL"
    }
```
So we'll probably want to add the `error` and `message` fields to this object.